ddtrace/_trace/tracer.py

```diff
@@ -283,9 +282,7 @@ def __init__(
             register_on_exit_signal(self._atexit)
 
         self._hooks = _hooks.Hooks()
-        atexit.register(self._atexit)
```

Comment:
With this change we no longer register `Tracer._atexit` and `Tracer._child_after_fork` when the tracer is initialized. This breaks all tests that generate spans in a subprocess where the main process does use ddtrace-run/import ddtrace.auto. This feels like a breaking change. Folks using only manual instrumentation may not receive spans when their processes exit and fork processes would not get a new instance of the AgentWriter/SpanAggreggators.

Reply:
In our docs we make it clear that user should use either the `ddtrace-run` wrapping command, or import `ddtrace.auto` asap. In these cases we still do the `atexit` and fork hooks.